File: 2966. Divide Array Into Arrays With Max Difference/divideArray.js
var divideArray = function (nums, k) {
  nums.sort((a, b) => a - b);

  const results = [];

  let counter = 0;
  let row = [];

  for (let i = 0; i <= nums.length; i++) {
    console.log(i, results, row);
    if (counter === 0) {
      console.log(row);
      results.push(row);
      counter = 0;
      row = [];
    }

    if (counter < 3) {
      row[counter] = nums[i];
    }

    counter++;
  }

  return results;
};

console.log(divideArray([1, 3, 4, 8, 7, 9, 3, 5, 1]), 2);

File: 2966. Divide Array Into Arrays With Max Difference/divideArray.go
package main

import (
	"fmt"
	"sort"
)

func divideArray(nums []int, k int) [][]int {
	sort.Ints(nums)
	results := [][]int{}

	for i := 0; i < len(nums); i += 3 {
		if nums[i+2]-nums[i] > k {
			return [][]int{}
		} else {
			results = append(results, nums[i:i+3])
		}
	}

	return results
}

func main() {
	// divideArray([]int{1, 3, 4, 8, 7, 9, 3, 5, 1}, 2)
	fmt.Println(divideArray([]int{1, 3, 4, 8, 7, 9, 3, 5, 1}, 2))
	fmt.Println(divideArray([]int{1, 3, 3, 2, 7, 3}, 3))
}
